layer/mounted_layer.go

```diff
@@ -155,6 +156,12 @@ func (rl *referencedRWLayer) Mount(mountLabel string) (string, error) {
 
 	if rl.activityCount > 0 {
 		rl.activityCount++
+		path, err := rl.mountedLayer.layerStore.GetMountPathFromDisk(rl.mountedLayer.name)
```

Comment:
I've moved {Get/Set}MountPath calls to mountedLayer, rather than referenceRWLayer. I've also added a pathCache to avoid filesystem lookup of the path.